polaris-service/src/main/java/org/apache/polaris/service/task/TableCleanupTaskHandler.java

```diff
@@ -158,6 +160,22 @@ public boolean handleTask(TaskEntity cleanupTask) {
         for (PolarisBaseEntity createdTask : createdTasks) {
           taskExecutor.addTaskHandlerContext(createdTask.getId(), CallContext.getCurrentContext());
         }
+
+        tableMetadata.snapshots().stream()
+            .flatMap(sn -> sn.allManifests(fileIO).stream())
+            // remove duplication
+            .collect(Collectors.toMap(ManifestFile::path, Function.identity(), (mf1, mf2) -> mf1))
+            .keySet()
+            .forEach(fileIO::deleteFile);
+        tableMetadata.snapshots().stream()
+            .map(Snapshot::manifestListLocation)
+            .forEach(fileIO::deleteFile);
```

Comment:
I think we've already handled the manifest list files.

Reply:
> I think we've already handled the manifest files and manifest list files.

Thank you for pointing that out! I just notice that these are already handled in the `ManifestFileCleanupTaskHandler`. I’ve removed the redundant deletion logic in the latest commit.